pkg/api/config.go

```diff
@@ -110,8 +110,9 @@ func validateTestStepConfiguration(fieldRoot string, input []TestStepConfigurati
 			validationErrors = append(validationErrors, fmt.Errorf("%s[%d].as: '%s' is not valid value, should be [a-zA-Z0-9_.-]", fieldRoot, num, test.As))
 		}
 
-		if len(test.Commands) == 0 {
-			validationErrors = append(validationErrors, fmt.Errorf("%s[%d].commands: is required", fieldRoot, num))
+		hasCommands, hasSteps := len(test.Commands) != 0, len(test.Steps) != 0
+		if !(hasCommands || hasSteps) || (hasCommands && hasSteps) {
+			validationErrors = append(validationErrors, fmt.Errorf("%s[%d]: either `commands` or `steps` (but not both) should be set", fieldRoot, num))
```

Comment:
Separate error messages are fine with me.

As for validation in general, as you said below, it is very incomplete. This was mostly a reminder to do it, I didn't want to spend too much time writing validation code  because I expect the configuration format to change.